components/motor/gpio/basic.go

```diff
@@ -177,8 +177,15 @@ func (m *Motor) setPWM(ctx context.Context, powerPct float64, extra map[string]i
 	var errs error
 	powerPct = math.Min(powerPct, m.maxPowerPct)
 	powerPct = math.Max(powerPct, -1*m.maxPowerPct)
+	powerPct = math.Max(powerPct, m.minPowerPct)
```

Comment:
just ran through this quickly with an example of -50% power and I think line 180 forces the powerPct to 0. this is assuming a min power of 0 and a max power of 100, meaning the power range is -100 to +100, assuming I am understanding the min and max powers correctly

Reply:
hmmm maybe i should add a greater than check here first.

Reply:
Another thing I could do is move the storage of powerPct and on to the parent SetPower if that makes this function less confusing, but let me check that `setPWM` is not called from anywhere else in `basic.go` later today.